src/browser/BrowserAction.cpp

```diff
@@ -501,6 +504,39 @@ QJsonObject BrowserAction::handleGetTotp(const QJsonObject& json, const QString&
     return buildResponse(action, message, newNonce);
 }
 
+QJsonObject BrowserAction::handleDeleteEntry(const QJsonObject& json, const QString& action)
+{
+    const QString nonce = json.value("nonce").toString();
+    const QString encrypted = json.value("message").toString();
+
+    if (!m_associated) {
+        return getErrorReply(action, ERROR_KEEPASS_ASSOCIATION_FAILED);
+    }
+
+    const QJsonObject decrypted = decryptMessage(encrypted, nonce);
+    if (decrypted.isEmpty()) {
+        return getErrorReply(action, ERROR_KEEPASS_CANNOT_DECRYPT_MESSAGE);
+    }
+
+    QString command = decrypted.value("action").toString();
+    if (command.isEmpty() || command.compare("delete-entry", Qt::CaseSensitive) != 0) {
+        return getErrorReply(action, ERROR_KEEPASS_INCORRECT_ACTION);
+    }
+
+    const auto uuid = decrypted.value("uuid").toString();
```

Comment:
I've been thinking, we really should sanitize the input here. We pass these values directly to some internal function calls without sanitizing them. Should at least determine if it is valid uuid length and character makeup.

Reply:
Got it. I guess there's no existing validation for it?

Reply:
There might be when you get to Entry::findEntryByUuid but we should catch any anomalies or malicious data well before then.

Reply:
I added some basic checks for UUID to `Tools::isValidUuid()`.